### DISCUSSION TITLE
firefox: cannot update itself (/opt install)

### DISCUSSION THREAD

**Discussion by Gerenuk** [netblue30/firejail#5034] [Q&A] [answered]
Is it possible that Firefox does not update itself when it is started through Firejail?
It would offer me to go to the download site and get the bz2-file, but it cannot automatically update itself as it usually could do(?)
If so, then a notice in the docs would be great or even a suggestion how to solve it. Otherwise the users run the risk of not noticing and having vulnerable browsers for a longer time.

**Comment by rusty-snake** [comment, marked as answer]
How did you installed firefox? Usually you install it with you package manager which is responsible for updating.

**Reply by Gerenuk** [reply]
I installed it from Mozilla directly, because then updates come in a bit faster. Any chance to make this version work again?
(in any case a warning in the documentation would be good)

**Reply by rusty-snake** [reply]
> I installed it from Mozilla directly

How?

**Reply by Gerenuk** [reply]
I think I just unpacked it into a directory. Automatic updates worked before I used firejail. Not sure how Mozilla does that.

**Reply by rusty-snake** [reply]
Which directory?

**Reply by Gerenuk** [reply]
in my case /opt/Firefox. But I assume it would work with any directory. Is there anything I could add to the config to make it work with a custom directory?

**Reply by rusty-snake** [reply]
See #3658.

**Reply by Gerenuk** [reply]
Oh thanks. I didn't realize /opt is not writeable. I guess it's safer if I keep it this way and update it myself Firefox when it is not running.
